<p>Passwords must not be logged.</p>

<h2>Noncompliant Code Example</h2>
<pre>
class MyClass {
    int     foo3(int value) {
    String password = "PASSWORD";
    log.info("This is not a valid test, because it contains a password: {}", password); // Noncompliant
    return 0;
  }
}
</pre>

<h2>Compliant Solution</h2>
<pre>
class MyClass {
  int     foo1() {
    log.error("This is a valid test");
  }

  void    foo2(int value) {
    log.warn("This is another valid test");
  }

  void    foo2(int value) {
    log.info("This is another valid test");
  }
}
</pre>
